lib/scarpe/download.rb

```diff
@@ -4,44 +4,124 @@
 require "openssl"
 require "nokogiri"
 
-class Scarpe::Widget
-  def download(url)
-    Thread.new do
-      begin
-        uri = URI(url)
-        http = Net::HTTP.new(uri.host, uri.port)
-        http.use_ssl = true
-        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+class Scarpe
+  class Widget
+    class ResponseWrapper
```

Comment:
note here we can avoid this gibrish like making struct and class if we use http-library gem ig it does provides a direct .header method but yeah it would add another gem

Reply:
If we were going to add another gem, I would instead suggest something that gets rid of the Thread.new (e.g. curb/libcurl, something that will do background download.) I'm fine with a DSL adapter layer, that's no big problem. But getting all the GUI calls out of background threads is a bigger deal.

For now, not gonna worry too much about it. I filed another bug about figuring out concurrency later, because download isn't the only Shoes API that will need sorting out.